from PyQt5 import QtCore, QtGui, QtWidgets
import sqlite3
from PyQt5.QtWidgets import QMessageBox

class Ui_BANK_DETAILS(object):
    def setupUi(self, BANK_DETAILS):
        BANK_DETAILS.setObjectName("BANK_DETAILS")
        BANK_DETAILS.resize(1368, 878)
        BANK_DETAILS.setMinimumSize(QtCore.QSize(1368, 878))
        BANK_DETAILS.setMaximumSize(QtCore.QSize(1368, 878))
        self.centralwidget = QtWidgets.QWidget(BANK_DETAILS)
        self.centralwidget.setObjectName("centralwidget")
        self.Bank_Details = QtWidgets.QFrame(self.centralwidget)
        self.Bank_Details.setGeometry(QtCore.QRect(0, 0, 1371, 841))
        self.Bank_Details.setStyleSheet("background-color: rgb(179, 179, 179);")
        self.Bank_Details.setFrameShape(QtWidgets.QFrame.StyledPanel)
        self.Bank_Details.setFrameShadow(QtWidgets.QFrame.Raised)
        self.Bank_Details.setObjectName("Bank_Details")
        self.f2 = QtWidgets.QFrame(self.Bank_Details)
        self.f2.setGeometry(QtCore.QRect(10, 10, 221, 781))
        self.f2.setStyleSheet("background-color:rgb(170, 85, 127)")
        self.f2.setFrameShape(QtWidgets.QFrame.StyledPanel)
        self.f2.setFrameShadow(QtWidgets.QFrame.Raised)
        self.f2.setObjectName("f2")
        self.l1 = QtWidgets.QLabel(self.Bank_Details)
        self.l1.setGeometry(QtCore.QRect(500, 130, 191, 51))
        font = QtGui.QFont()
        font.setPointSize(12)
        font.setBold(True)
        font.setWeight(75)
        self.l1.setFont(font)
        self.l1.setObjectName("l1")
        self.combo1 = QtWidgets.QComboBox(self.Bank_Details)
        self.combo1.setGeometry(QtCore.QRect(690, 130, 231, 41))
        font = QtGui.QFont()
        font.setPointSize(10)
        font.setBold(True)
        font.setWeight(75)
        self.combo1.setFont(font)
        self.combo1.setStyleSheet("\n"
"background-color:rgb(200, 127, 155);")
        self.combo1.setObjectName("combo1")
        self.label = QtWidgets.QLabel(self.Bank_Details)
        self.label.setGeometry(QtCore.QRect(390, 235, 111, 41))
        self.label.setText("")
        self.label.setObjectName("label")
        self.label_2 = QtWidgets.QLabel(self.Bank_Details)
        self.label_2.setGeometry(QtCore.QRect(500, 240, 121, 41))
        font = QtGui.QFont()
        font.setPointSize(12)
        font.setBold(True)
        font.setWeight(75)
        self.label_2.setFont(font)
        self.label_2.setObjectName("label_2")
        self.lineEdit = QtWidgets.QLineEdit(self.Bank_Details)
        self.lineEdit.setGeometry(QtCore.QRect(690, 240, 231, 41))
        font = QtGui.QFont()
        font.setPointSize(9)
        font.setBold(True)
        font.setWeight(75)
        self.lineEdit.setFont(font)
        self.lineEdit.setStyleSheet("background-color:rgb(200, 127, 155);")
        self.lineEdit.setObjectName("lineEdit")
        self.label_3 = QtWidgets.QLabel(self.Bank_Details)
        self.label_3.setGeometry(QtCore.QRect(500, 350, 111, 41))
        font = QtGui.QFont()
        font.setPointSize(12)
        font.setBold(True)
        font.setWeight(75)
        self.label_3.setFont(font)
        self.label_3.setObjectName("label_3")
        self.le2 = QtWidgets.QLineEdit(self.Bank_Details)
        self.le2.setGeometry(QtCore.QRect(690, 350, 231, 41))
        font = QtGui.QFont()
        font.setBold(True)
        font.setWeight(75)
        self.le2.setFont(font)
        self.le2.setStyleSheet("background-color:rgb(200, 127, 155);")
        self.le2.setObjectName("le2")
        self.pb1 = QtWidgets.QPushButton(self.Bank_Details)
        self.pb1.setGeometry(QtCore.QRect(560, 450, 191, 51))
        font = QtGui.QFont()
        font.setPointSize(12)
        font.setBold(True)
        font.setWeight(75)
        self.pb1.setFont(font)
        self.pb1.setCursor(QtGui.QCursor(QtCore.Qt.PointingHandCursor))
        self.pb1.setStyleSheet("backgroung-color:rgb(255, 255, 255);")
        self.pb1.setObjectName("pb1")
        self.label_4 = QtWidgets.QLabel(self.Bank_Details)
        self.label_4.setGeometry(QtCore.QRect(570, 30, 241, 61))
        font = QtGui.QFont()
        font.setPointSize(20)
        font.setBold(True)
        font.setWeight(75)
        self.label_4.setFont(font)
        self.label_4.setObjectName("label_4")
        BANK_DETAILS.setCentralWidget(self.centralwidget)
        self.menubar = QtWidgets.QMenuBar(BANK_DETAILS)
        self.menubar.setGeometry(QtCore.QRect(0, 0, 1368, 27))
        self.menubar.setObjectName("menubar")
        BANK_DETAILS.setMenuBar(self.menubar)
        self.statusbar = QtWidgets.QStatusBar(BANK_DETAILS)
        self.statusbar.setObjectName("statusbar")
        BANK_DETAILS.setStatusBar(self.statusbar)

        self.pb1.clicked.connect(self.bankdetails)
        self.subName()

        self.retranslateUi(BANK_DETAILS)
        QtCore.QMetaObject.connectSlotsByName(BANK_DETAILS)

    def retranslateUi(self, BANK_DETAILS):
        _translate = QtCore.QCoreApplication.translate
        BANK_DETAILS.setWindowTitle(_translate("BANK_DETAILS", "Bank_Details"))
        self.l1.setText(_translate("BANK_DETAILS", "BANK NAME:"))
        self.label_2.setText(_translate("BANK_DETAILS", "IFSC CODE:"))
        self.label_3.setText(_translate("BANK_DETAILS", "BRANCH:"))
        self.pb1.setText(_translate("BANK_DETAILS", "SUBMIT"))
        self.label_4.setText(_translate("BANK_DETAILS", "Bank Details"))

    def bankdetails(self):
        br = self.le2.text().lstrip().rstrip()
        ifsc = self.lineEdit.text().lstrip().rstrip()
        bn = self.combo1.currentText()

        con = sqlite3.connect("UMRS.db")
        cur = con.cursor()
        q = "insert into Bank_Details values (?,?,?)"
        cur.execute(q,(bn,ifsc,br))
        con.commit()

        msg = QMessageBox()
        msg.setIcon(QMessageBox.Information)
        msg.setText("1 Bank Details added")
        msg.setWindowTitle("Band Details")
        msg.show()
        msg.exec_()

    def subName(self):
        bn = self.combo1.currentText()
        con = sqlite3.connect("UMRS.db")
        cur = con.cursor()
        q1 = "select BankName from Bank_Name"
        cur.execute(q1,(bn))
        c = cur.fetchall()
        for i in c:
            self.combo1.addItem(i[0])


if __name__ == "__main__":
    import sys
    app = QtWidgets.QApplication(sys.argv)
    BANK_DETAILS = QtWidgets.QMainWindow()
    ui = Ui_BANK_DETAILS()
    ui.setupUi(BANK_DETAILS)
    BANK_DETAILS.show()
    sys.exit(app.exec_())
